src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        user = new User();
+        user.setLogin("Tretiak");
+        user.setPassword("123456");
+        user.setAge(20);
+    }
+
+    @Test
+    void register_noExistUserLogin_Ok() {
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    void register_containsUserLogin_NotOk() {
+        Storage.people.add(user);
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("User already exist!");
+    }
+
+    @Test
+    void register_nullUserLogin_NotOk() {
+        user.setLogin(null);
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Login can't be null!");
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        user = null;
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("User can't be null!");
+    }
+
+    @Test
+    void register_smallPassword_NotOk() {
+        user.setPassword("123");
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Password less than 6 symbols!");
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        user.setPassword(null);
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Password can't be null!");
+    }
+
+    @Test
+    void register_lowAge_NotOk() {
+        user.setAge(15);
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Age less than 18 years old!");

```

Comment:
Look in the [common mistake](https://mate-academy.github.io/jv-program-common-mistakes/java-core/junit/registration-validation#:~:text=If%20you%20are%20expecting%20exception%20to%20be%20thrown%20in%20the%20test%2C%20you%20can%20do%20it%20this%20way)